astropy/modeling/separable.py

```diff
@@ -0,0 +1,290 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""
+Functions to determine if a model is separable, i.e.
+if the model outputs are independent.
+
+It analyzes ``n_inputs``, ``n_outputs`` and the operators
+in a compound model by stepping through the transforms
+and creating a ``coord_matrix`` of shape (``n_outputs``, ``n_inputs``).
+
+
+Each modeling operator is represented by a function which
+takes two simple models (or two ``coord_matrix`` arrays) and
+returns an array of shape (``n_outputs``, ``n_inputs``).
+
+"""
+
+import numpy as np
+
+from .core import Model, _CompoundModel, ModelDefinitionError
+from .mappings import Mapping
+from .models import Shift, Scale, Rotation2D, Polynomial2D
```

Comment:
Looks like there are only used in examples in the docstrings. Would it be possible to rework those examples to contain all the necessary imports there, otherwise none of those would work as copy pasted for the users.